internal/mode/static/state/graph/namespace.go

```diff
@@ -0,0 +1,43 @@
+package graph
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// buildReferencedNamespaces returns a map of all the Namespace resources in the current clusterState with a label
+// that matches any of the Gateway Listener's label selector.
+func buildReferencedNamespaces(clusterNamespaces map[types.NamespacedName]*v1.Namespace,
+	gw *Gateway,
+) map[types.NamespacedName]*v1.Namespace {
+	referencedNamespaces := make(map[types.NamespacedName]*v1.Namespace)
+	for name, ns := range clusterNamespaces {
+		if isNamespaceReferenced(ns, gw) {
+			referencedNamespaces[name] = ns
+		}
+	}
+	if len(referencedNamespaces) == 0 {
+		return nil
+	}
+	return referencedNamespaces
```

Comment:
Do we have to return nil here? Or can we just return the empty map?

Reply:
I was going off of how the ReferencedSecrets were implemented where if there were no "ResolvedSecrets", it would return nil. [Here](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/internal/mode/static/state/graph/secret.go#L70C3-L70C3).